test/test_console_tool.py

```diff
@@ -566,6 +566,183 @@ def test_get_bucket_show_size(self):
             '''
             self._run_command(['get-bucket', '--showSize', 'my-bucket'], expected_stdout, '', 0)
 
+    def test_get_bucket_with_versions(self):
+        self._authorize_account()
+        self._create_my_bucket()
+
+        # Put many versions of a file into the test bucket. Unroll the loop here for convenience.
```

Comment:
Why unroll the loop?

Reply:
It's more convenient to structure the tests here through lexical code cloning rather than algorithmic duplication, especially in some of the tests that create varied hierarchical structures. While it's certainly possible to rewrite using loops and interpolation logic, I would argue that it's simpler and easier to maintain an unrolled loop. Given that it's Python and not a systems language, it's probably not any more efficient to unroll the loop; I presume that the VM overhead will dwarf any loop execution overhead, so efficiency isn't really a concern.

Reply:
I agree that it's all a matter of code style: what's the easiest for a reader of the code to understand.  I'm fine with the code either way.

Efficiency is definitely not a concern here.  It's only doing a small number of files.